addons/web/controllers/main.py

```diff
@@ -325,18 +325,31 @@ def set_cookie_and_redirect(redirect_url):
     return redirect
 
 def clean_action(action, env):
-    action.setdefault('flags', {})
     action_type = action.setdefault('type', 'ir.actions.act_window_close')
     if action_type == 'ir.actions.act_window':
         action = fix_view_modes(action)
 
-    # When returning an action, only few information are really usefull
-    return {
+    # When returning an action, keep only relevant fields/properties
+    readable_fields = env[action['type']]._get_readable_fields()
+    action_type_fields = env[action['type']]._fields.keys()
+
+    cleaned_action = {
         field: value
         for field, value in action.items()
-        if field in env[action['type']]._get_readable_fields()
+        # keep allowed fields and custom properties fields
+        if field in readable_fields or field not in action_type_fields
     }
 
+    # Warn about custom properties fields, because use is discouraged
+    action_name = action.get('name') or action
+    custom_properties = action.keys() - readable_fields - action_type_fields
+    if custom_properties:
+        _logger.warning("Action %s contains custom properties %s, passing them "
```

Comment:
Maybe use `%r` so that if we have an actual action name it's properly quoted (and easier to identify) in the logging message?

Also maybe explicitly format the `custom_properties`? As is, it's going to show them as a set literal e.g.

    Action this is my action contains custom properties {'infos', 'flags'}, passing them via the `params` or `context` properties is recommended instead.